guides/development-workflow.md

```diff
@@ -0,0 +1,342 @@
+
+```
+feature/{STORY ID}-{CONCISE TITLE}
+fix/{STORY/TICKET ID}-{CONCISE TITLE}
+chore/{STORY ID}-{CONCISE TITLE}
+hotfix/{TICKET ID}-{CONCISE TITLE}
+```
+
+**Examples:**
+
+```
+feature/523797477-add-logging-to-registration
+fix/423797477-logging-happens-in-both-environments
+chore/1934729239-reduce-caching-for-contact-details
+hotfix/253947623-remove-breaking-change-to-repair-creation
+```
+
+#### Update your branch with the latest changes on develop (rebasing)
+
+When working in a team other pull requests will often get merged in before yours is finished  It's important to make sure you're developing against the latest version of the code base as much as possible.
+
+You can do this by rebasing your work on top of the `develop` branch.
+
+```
+git checkout <branch-name>
+git rebase origin/develop
+git push --force-with-lease
+```
+
+During the rebase there may be conflicts which should be resolved at this point. If in doubt talk to the other developer who you're conflicting with and reason out what the correct state should be.
+
+Force pushing in this situation is the only time it should be necessary, adding `--force-with-lease` ensures you don't accidentally overwrite someone else's additions.
+
+## Committing
+
+Writing a good commit message serves three important purposes:
+
+* to speed up the review process
+* to encourage individual commits to be small and well bounded
+* to help future maintainers (it could be you) in say 3 years time, to find out why a particular change was made and what context was relevant at the time
+
+**Convention:**
+
+```
+(feature): when we are adding something new
+(fix): when we are fixing something existing
+(refactor): when we are changing the implementation of something existing without changing its behaviour
+(chore): catch all for when none of the above apply and there is no immediate user need eg. gem/package updates
+```
+
+**Examples:**
+
+```
+# Poor
+Added repairs
+
+# Great
+(feature): Job postings are now displayed on the landing page
+
+* Getting the Job postings out of the external API proved difficult but I think the solution we have will allow us to easily upgrade to the new release they have planned: <link>
+```
+
+```
+# Poor
+Fixed the form
+
+# Great
+(fix): Food order form works with special characters
+
+* Previously the form failed to send if the starter value included any of the following characters: `!@£$%^&`
+* Users should be able to use these characters and after a short investigation I found this new bug in PHP: <link>
+* As a short term workaround until the bug is fixed I am stripping out any of these characters if they are present.
+```
+
+```
+# Poor
+Gem upgrade
+
+# Great
+(chore): Upgrade widget to v1.2.1
+
+* This widget now includes a patch that improves resilience: <link>
+* We are still unable to upgrade to v2.x as we need to do some migration work first
+```
+---
+
+## Code review
```

Comment:
We seem to have lost the link to [Thoughtbot's code review guide](https://github.com/thoughtbot/guides/tree/master/code-review)

We don't seem to have any guidance on how to do a good code review. A list of things to check when you're reviewing code might be helpful